needle.pm

```diff
@@ -141,17 +136,15 @@ sub save {
     close $fh;
 }
 
-sub unregister {
-    my ($self, $reason) = @_;
+sub unregister ($self, $reason = 'unknown reason') {
```

Comment:
As mentioned many times before, setting a default here is not equivalent to the previous code. Note that this function is invoked by the test distribution.

Reply:
I will never remember that, sorry

Reply:
like this with `$reason = undef` and keeping the old `… || 'unknown reason'`?